Subject: [PATCH] Fix FFmpeg build failure
---
Index: zconf.h
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/zconf.h b/zconf.h
--- a/zconf.h	(revision 00161eff1de25e9eed56bffdbe58f2c07ca16e51)
+++ b/zconf.h	(date 1739496977666)
@@ -436,11 +436,11 @@
    typedef unsigned long z_crc_t;
 #endif
 
-#ifdef HAVE_UNISTD_H    /* may be set to #if 1 by ./configure */
+#if defined(HAVE_UNISTD_H) && HAVE_UNISTD_H
 #  define Z_HAVE_UNISTD_H
 #endif
 
-#ifdef HAVE_STDARG_H    /* may be set to #if 1 by ./configure */
+#if defined(HAVE_STDARG_H) && HAVE_STDARG_H
 #  define Z_HAVE_STDARG_H
 #endif
 
