Subject: [PATCH] Fix CMake build
---
Index: CMakeLists.txt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	(revision e1774cd77f471443541596e09078e78fdc342e4f)
+++ b/CMakeLists.txt	(date 1745978460868)
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.8.12)
+cmake_minimum_required(VERSION 3.6)
 project(libogg)
 
 # Required modules
