diff -urpN glib-2.42.0.orig/config.h.win32 glib-2.42.0/config.h.win32
--- glib-2.42.0.orig/config.h.win32	2014-05-03 13:34:53 -0700
+++ glib-2.42.0/config.h.win32	2014-05-03 13:35:27 -0700
@@ -207,10 +207,8 @@
 /* Define to 1 if you have the `hasmntopt' function. */
 /* #undef HAVE_HASMNTOPT */

-/* Target the Windows XP API */
-#ifndef _WIN32_WINNT
-#define _WIN32_WINNT 0x0501
-#endif
+/* Target the Windows Vista API */
+#define _WIN32_WINNT 0x0600

 #if _WIN32_WINNT >= 0x0600
 /* if_indextoname() and if_nametoindex() are available on Windows Vista/Server
