diff -ruN --strip-trailing-cr cairo/src/cairo-cache-private.h build/Win32/cairo/src/cairo-cache-private.h
--- cairo/src/cairo-cache-private.h	2015-03-10 18:21:07.000000000 -0400
+++ build/Win32/cairo/src/cairo-cache-private.h	2015-12-04 10:16:00.304925900 -0500
@@ -84,7 +84,7 @@
  * not be initialized if so desired.
  **/
 typedef struct _cairo_cache_entry {
-    unsigned long hash;
+    unsigned long long hash;
     unsigned long size;
 } cairo_cache_entry_t;
 
diff -ruN --strip-trailing-cr cairo/src/cairo-scaled-font.c build/Win32/cairo/src/cairo-scaled-font.c
--- cairo/src/cairo-scaled-font.c	2015-03-10 18:21:07.000000000 -0400
+++ build/Win32/cairo/src/cairo-scaled-font.c	2015-12-04 10:16:00.315176900 -0500
@@ -2852,7 +2852,7 @@
     if (unlikely (page == NULL))
 	return _cairo_error (CAIRO_STATUS_NO_MEMORY);
 
-    page->cache_entry.hash = (unsigned long) scaled_font;
+    page->cache_entry.hash = (unsigned long long) scaled_font;
     page->cache_entry.size = 1; /* XXX occupancy weighting? */
     page->num_glyphs = 0;
 