diff -urpN pango-1.36.8.orig/pango/pangowin32-fontmap.c pango-1.36.8/pango/pangowin32-fontmap.c
--- pango-1.36.8.orig/pango/pangowin32-fontmap.c	Thu Jul 31 23:49:36 2014
+++ pango-1.36.8/pango/pangowin32-fontmap.c	Sat Nov 22 10:23:07 2014
@@ -243,28 +243,27 @@ synthesize_foreach (gpointer key,
   while (p)
     {
       PangoWin32Face *win32face = p->data;
+      PangoWeight weight = pango_font_description_get_weight (win32face->description);
+      PangoStyle style = pango_font_description_get_style (win32face->description);
 
-      /* Don't synthesize anything unless it's a monospace, serif, or sans font */
-      if (!((win32face->logfontw.lfPitchAndFamily & 0xF0) == FF_MODERN ||
-	    (win32face->logfontw.lfPitchAndFamily & 0xF0) == FF_ROMAN ||
-	    (win32face->logfontw.lfPitchAndFamily & 0xF0) == FF_SWISS))
-	return;
-
-      if (pango_font_description_get_weight (win32face->description) == PANGO_WEIGHT_NORMAL &&
-	  pango_font_description_get_style (win32face->description) == PANGO_STYLE_NORMAL)
-	variant[NORMAL] = win32face;
-
-      if (pango_font_description_get_weight (win32face->description) > PANGO_WEIGHT_NORMAL &&
-	  pango_font_description_get_style (win32face->description) == PANGO_STYLE_NORMAL)
-	variant[BOLDER] = win32face;
-
-      if (pango_font_description_get_weight (win32face->description) == PANGO_WEIGHT_NORMAL &&
-	  pango_font_description_get_style (win32face->description) >= PANGO_STYLE_OBLIQUE)
-	variant[SLANTED] = win32face;
-
-      if (pango_font_description_get_weight (win32face->description) > PANGO_WEIGHT_NORMAL &&
-	  pango_font_description_get_style (win32face->description) >= PANGO_STYLE_OBLIQUE)
-	variant[BOLDER+SLANTED] = win32face;
+      gboolean weight_is_normal =
+        weight > (PANGO_WEIGHT_LIGHT + PANGO_WEIGHT_NORMAL) / 2 &&
+        weight <= (PANGO_WEIGHT_NORMAL + PANGO_WEIGHT_SEMIBOLD) / 2;
+
+      gboolean weight_is_bolder =
+        weight > (PANGO_WEIGHT_NORMAL + PANGO_WEIGHT_SEMIBOLD) / 2;
+
+      if (weight_is_normal && style == PANGO_STYLE_NORMAL)
+        variant[NORMAL] = win32face;
+
+      else if (weight_is_bolder && style == PANGO_STYLE_NORMAL)
+        variant[BOLDER] = win32face;
+
+      else if (weight_is_normal && style >= PANGO_STYLE_OBLIQUE)
+        variant[SLANTED] = win32face;
+
+      else if (weight_is_bolder && style >= PANGO_STYLE_OBLIQUE)
+        variant[BOLDER+SLANTED] = win32face;
 
       p = p->next;
     }
