From ad9d81d3191acd886562d9580511b8b5dd4362ee Mon Sep 17 00:00:00 2001
From: Seungha Yang <seungha.yang@navercorp.com>
Date: Sun, 12 May 2019 18:33:32 +0900
Subject: [PATCH] meson: Don't try to find gio-unix on Windows

---
 meson.build | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/meson.build b/meson.build
index 1a6bdd66c..91be429b1 100644
--- a/meson.build
+++ b/meson.build
@@ -251,8 +251,11 @@ glib_deps = [dependency('glib-2.0', version : glib_req, fallback: ['glib', 'libg
              dependency('gobject-2.0', fallback: ['glib', 'libgobject_dep'])]
 # GIO is used by the GIO plugin, and by the TCP, SDP, and RTSP plugins
 gio_dep = dependency('gio-2.0', fallback: ['glib', 'libgio_dep'])
-giounix_dep = dependency('gio-unix-2.0', version : glib_req, required : host_system != 'windows',
-                         fallback: ['glib', 'libgiounix_dep'])
+giounix_dep = dependency('', required: false)
+if host_system != 'windows'
+  giounix_dep = dependency('gio-unix-2.0', version : glib_req,
+                           fallback: ['glib', 'libgiounix_dep'])
+endif
 gmodule_dep = dependency('gmodule-no-export-2.0',
                          fallback: ['glib', 'libgmodule_dep'])
 
-- 
2.17.1

