From 7282d3289c6f6e2cc21e2c2b554c33af00ac3474 Mon Sep 17 00:00:00 2001
From: Nobuyoshi Nakada <nobu@ruby-lang.org>
Date: Mon, 22 Dec 2014 17:14:40 +0900
Subject: [PATCH] x86: MSVC does not support Complex type

---
 src/x86/ffitarget.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/x86/ffitarget.h b/src/x86/ffitarget.h
index 580522f..8c1dcac 100644
--- a/src/x86/ffitarget.h
+++ b/src/x86/ffitarget.h
@@ -50,7 +50,9 @@
 #endif
 
 #define FFI_TARGET_SPECIFIC_STACK_SPACE_ALLOCATION
+#ifndef _MSC_VER
 #define FFI_TARGET_HAS_COMPLEX_TYPE
+#endif
 
 /* ---- Generic type definitions ----------------------------------------- */
 
-- 
1.9.5.github.0

