From f9ddb70a8c64a36032a9bab79bd5e065192143bb Mon Sep 17 00:00:00 2001
From: Danny 'Guru' Forghieri <daniele.forghieri@gmail.com>
Date: Mon, 21 Nov 2022 12:17:36 +0100
Subject: [PATCH] libpkgconf: add defines to unbreak build with VS2013

---
 libpkgconf/bsdstubs.c   | 1 +
 libpkgconf/libpkgconf.h | 8 ++++++++
 2 files changed, 9 insertions(+)

diff --git a/libpkgconf/bsdstubs.c b/libpkgconf/bsdstubs.c
index aebf831..3c6399f 100644
--- a/libpkgconf/bsdstubs.c
+++ b/libpkgconf/bsdstubs.c
@@ -14,6 +14,7 @@
 
 #include <libpkgconf/stdinc.h>
 #include <libpkgconf/bsdstubs.h>
+#include <libpkgconf/libpkgconf.h>
 #include <libpkgconf/config.h>
 
 #if HAVE_DECL_STRNDUP
diff --git a/libpkgconf/libpkgconf.h b/libpkgconf/libpkgconf.h
index 28d1bd4..f435294 100644
--- a/libpkgconf/libpkgconf.h
+++ b/libpkgconf/libpkgconf.h
@@ -29,6 +29,13 @@
 #include <libpkgconf/iter.h>
 #include <libpkgconf/bsdstubs.h>
 
+/* Macro for building with vs2013 */
+#if defined _MSC_VER && _MSC_VER <= 1800
+#define inline         __inline
+#define snprintf       _snprintf
+#define __func__       __FUNCTION__
+#endif
+
 #ifdef __cplusplus
 extern "C" {
 #endif
(END)