From d79a502475e7dbd984e8e24cbcea564dc5631e3b Mon Sep 17 00:00:00 2001
From: Ignacio Casal Quinteiro <icq@gnome.org>
Date: Mon, 17 Oct 2016 17:49:55 +0200
Subject: [PATCH 1/2] websocket-connection: log when getting a pong message

---
 libsoup/soup-websocket-connection.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libsoup/soup-websocket-connection.c b/libsoup/soup-websocket-connection.c
index 8901941..4625716 100644
--- a/libsoup/soup-websocket-connection.c
+++ b/libsoup/soup-websocket-connection.c
@@ -586,6 +586,7 @@ process_contents (SoupWebsocketConnection *self,
 			receive_ping (self, payload, payload_len);
 			break;
 		case 0x0A:
+			g_debug ("received pong message");
 			break;
 		default:
 			g_debug ("received unsupported control frame: %d", (int)opcode);
-- 
2.9.3

