From b06107e5e3e9fef9b42699b6179456ac1804264a Mon Sep 17 00:00:00 2001
From: Chun-wei Fan <fanchunwei@src.gnome.org>
Date: Wed, 6 Apr 2016 17:22:23 +0800
Subject: [PATCH] gettext-tools/src/x-lua.c: Fix C99ism

---
 gettext-tools/src/x-lua.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/gettext-tools/src/x-lua.c b/gettext-tools/src/x-lua.c
index 5050838..ff590be 100644
--- a/gettext-tools/src/x-lua.c
+++ b/gettext-tools/src/x-lua.c
@@ -281,9 +281,9 @@ phase2_getc ()
 
           if (c == '[')
             {
+              int esigns = 0;
               c = phase1_getc ();
 
-              int esigns = 0;
               while (c == '=')
                 {
                   esigns++;
@@ -704,10 +704,11 @@ phase3_get (token_ty *tp)
           break;
 
         case '[':
+          {
+          int esigns = 0;
           c = phase1_getc ();
 
           /* Count the number of equal signs.  */
-          int esigns = 0;
           while (c == '=')
             {
               esigns++;
@@ -739,10 +740,10 @@ phase3_get (token_ty *tp)
 
               if (c == ']')
                 {
+                  int esigns2 = 0;
                   c = phase1_getc ();
 
                   /* Count the number of equal signs.  */
-                  int esigns2 = 0;
                   while (c == '=')
                     {
                       esigns2++;
@@ -792,6 +793,7 @@ phase3_get (token_ty *tp)
                 }
             }
           break;
+          }
 
         case ']':
           tp->type = token_type_rbracket;
-- 
2.7.1.windows.2

