diff -Naur gettext-orig/gettext-runtime/gnulib-lib/memchr.c gettext/gettext-runtime/gnulib-lib/memchr.c
--- gettext-orig/gettext-runtime/gnulib-lib/memchr.c    2015-12-27 15:08:01.000000000 -0800
+++ gettext/gettext-runtime/gnulib-lib/memchr.c 2018-05-19 13:06:58.868356900 -0700
@@ -53,7 +53,7 @@
 #ifndef weak_alias
 # define __memchr memchr
 #endif
-
+#if (!defined(_MSC_VER) || (_MSC_VER <= 1912))
 /* Search no more than N bytes of S for C.  */
 void *
 __memchr (void const *s, int c_in, size_t n)
@@ -167,6 +167,7 @@

   return NULL;
 }
+#endif
 #ifdef weak_alias
 weak_alias (__memchr, BP_SYM (memchr))
 #endif
diff -Naur gettext-orig/gettext-runtime/gnulib-lib/memmove.c gettext/gettext-runtime/gnulib-lib/memmove.c
--- gettext-orig/gettext-runtime/gnulib-lib/memmove.c   2015-12-27 15:08:01.000000000 -0800
+++ gettext/gettext-runtime/gnulib-lib/memmove.c        2018-05-19 13:06:36.826554000 -0700
@@ -6,7 +6,7 @@
 #include <config.h>

 #include <stddef.h>
-
+#if (!defined(_MSC_VER) || (_MSC_VER <= 1912))
 void *
 memmove (void *dest0, void const *source0, size_t length)
 {
@@ -24,3 +24,4 @@
     }
   return dest0;
 }
+#endif
\ No newline at end of file
diff -Naur gettext-orig/gettext-tools/libgettextpo/memchr.c gettext/gettext-tools/libgettextpo/memchr.c
--- gettext-orig/gettext-tools/libgettextpo/memchr.c    2015-12-27 15:09:03.000000000 -0800
+++ gettext/gettext-tools/libgettextpo/memchr.c 2018-05-19 13:07:48.873531200 -0700
@@ -53,7 +53,7 @@
 #ifndef weak_alias
 # define __memchr memchr
 #endif
-
+#if (!defined(_MSC_VER) || (_MSC_VER <= 1912))
 /* Search no more than N bytes of S for C.  */
 void *
 __memchr (void const *s, int c_in, size_t n)
@@ -167,6 +167,7 @@

   return NULL;
 }
+#endif
 #ifdef weak_alias
 weak_alias (__memchr, BP_SYM (memchr))
 #endif