diff --git a/CMakeLists.txt b/CMakeLists.txt
index 845ddf0..08f9487 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -181,10 +181,6 @@ foreach(flag ${test_flags})
   endif (${test_c_flag})
 endforeach(flag ${test_flags})
 
-if(MSVC)
-    set(CMAKE_DEBUG_POSTFIX "d")
-endif()
-
 option(ld-version-script "Enable linker version script" ON)
 # Check if LD supports linker scripts.
 file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/conftest.map" "VERS_1 {
