diff -ru cur/headers/meson.build new/headers/meson.build
--- cur/headers/meson.build	2019-03-11 23:33:15.000000000 +0100
+++ new/headers/meson.build	2019-07-15 22:51:15.985736800 +0200
@@ -21,7 +21,7 @@
 
 if not meson.is_cross_build() and get_option('introspection')
   noinst_lib = shared_library('noinst',
-                              headers,
+                              sources: [ headers, 'win_dll.c', ],
                               install: false)
 
   gnome.generate_gir(noinst_lib,
