From fd7c02a4dc49780de77d9222f044a704a5f54e92 Mon Sep 17 00:00:00 2001
From: Ignacio Casal Quinteiro <qignacio@amazon.com>
Date: Tue, 19 Oct 2021 15:03:51 +0200
Subject: [PATCH] libpcre: add a fallback url

---
 subprojects/libpcre.wrap | 1 +
 1 file changed, 1 insertion(+)

diff --git a/subprojects/libpcre.wrap b/subprojects/libpcre.wrap
index e27a198ec..921a627c7 100644
--- a/subprojects/libpcre.wrap
+++ b/subprojects/libpcre.wrap
@@ -6,6 +6,7 @@ source_hash = 51679ea8006ce31379fb0860e46dd86665d864b5020fc9cd19e71260eef4789d
 patch_filename = pcre_8.37-2_patch.zip
 patch_url = https://wrapdb.mesonbuild.com/v2/pcre_8.37-2/get_patch
 patch_hash = 6b80f72385e1bf06721e26fbc83aced576e9c0d3182d86a55dd173a04050fe26
+source_fallback_url = https://github.com/wingtk/gvsbuild/releases/download/untagged-7d774977c3a85aa9fc80/pcre-8.37.tar.bz2
 
 [provide]
 libpcre = pcre_dep
-- 
2.25.1

