diff --git a/subprojects/pcre.wrap b/subprojects/pcre.wrap
index a6b07b961..90b6d9b20 100644
--- a/subprojects/pcre.wrap
+++ b/subprojects/pcre.wrap
@@ -6,6 +6,7 @@ source_hash = 51679ea8006ce31379fb0860e46dd86665d864b5020fc9cd19e71260eef4789d
 patch_filename = pcre_8.37-4_patch.zip
 patch_url = https://wrapdb.mesonbuild.com/v2/pcre_8.37-4/get_patch
 patch_hash = c957f42da6f6378300eb8a18f4a5cccdb8e2aada51a703cac842982f9f785399
+source_fallback_url = https://github.com/wingtk/gvsbuild/releases/download/PCRE-8.37/pcre-8.37.tar.bz2
 
 [provide]
 libpcre = pcre_dep
