Subject: [PATCH] remove dll export check
---
Index: girepository/girepository.c
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/girepository/girepository.c b/girepository/girepository.c
--- a/girepository/girepository.c	(revision 0a7eb121fc19118534753addeae476a689d83e28)
+++ b/girepository/girepository.c	(date 1743954688020)
@@ -140,8 +140,6 @@
 
 static HMODULE girepository_dll = NULL;
 
-#ifdef DLL_EXPORT
-
 BOOL WINAPI DllMain (HINSTANCE hinstDLL, DWORD fdwReason, LPVOID lpvReserved);
 
 BOOL WINAPI
@@ -155,7 +153,6 @@
   return TRUE;
 }
 
-#endif /* DLL_EXPORT */
 #endif /* G_PLATFORM_WIN32 */
 
 #ifdef __APPLE__
