--- cogl-1.22.0.orig/cogl/driver/gl/cogl-framebuffer-gl.c	2015-08-07 00:16:30.000000000 +0900
+++ cogl-1.22.0/cogl/driver/gl/cogl-framebuffer-gl.c	2016-08-22 15:59:29.094093300 +0900
@@ -810,7 +810,7 @@ _cogl_offscreen_gl_allocate (CoglOffscre
 
   _COGL_RETURN_VAL_IF_FAIL (offscreen->texture_level <
                             _cogl_texture_get_n_levels (offscreen->texture),
-                            NULL);
+                            FALSE);
 
   _cogl_texture_get_level_size (offscreen->texture,
                                 offscreen->texture_level,
