diff -Naur gettext-orig/gettext-runtime/intl/CMakeLists.txt gettext-mod/gettext-runtime/intl/CMakeLists.txt
--- gettext-orig/gettext-runtime/intl/CMakeLists.txt	2012-01-21 21:10:25.000000000 -0700
+++ gettext-mod/gettext-runtime/intl/CMakeLists.txt	2015-12-02 16:23:51.903960200 -0700
@@ -55,9 +55,6 @@
 
 add_library(intl ${LIBRARY_TYPE} ${INTL_SRCS})
 target_link_libraries(intl ${ICONV_LIBRARIES})
-if(MSVC)
-set_target_properties(intl PROPERTIES OUTPUT_NAME "libintl")
-endif(MSVC)
 
 if(WINCE)
     target_link_libraries(intl ${WCECOMPAT_LIBRARIES})
