diff --git a/meson.build b/meson.build
index 5eaae563..b31cb644 100644
--- a/meson.build
+++ b/meson.build
@@ -171,7 +171,7 @@ if host_system != 'windows'
 endif

 # User options
-if get_option('debug')
+if get_option('ffi-debug')
   ffi_conf.set('FFI_DEBUG')
 endif
 if not get_option('raw_api')
diff --git a/meson_options.txt b/meson_options.txt
index 7e5bede4..2c7f517a 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -1,5 +1,5 @@
 # Toggle this if you want extra debugging
-option('debug', type : 'boolean', value : false)
+option('ffi-debug', type : 'boolean', value : false)
 # Toggle this if you do not want support for aggregate types
 option('structs', type : 'boolean', value : true)
 # Toggle this if you do not want support for the raw API
diff --git a/src/meson.build b/src/meson.build
index 9a9feba1..ed5c1840 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -8,7 +8,7 @@ ffi_c_sources = [

 ffi_asm_sources = []

-if get_option('debug')
+if get_option('ffi-debug')
   ffi_c_sources += ['debug.c']
 endif
