Subject: [PATCH] Fix missing ;
---
Index: gi/pygi-resulttuple.c
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/gi/pygi-resulttuple.c b/gi/pygi-resulttuple.c
--- a/gi/pygi-resulttuple.c	(revision 5d3a375c5c683d7bcd3669c1d9957cfd804f97b5)
+++ b/gi/pygi-resulttuple.c	(date 1760405716194)
@@ -335,7 +335,7 @@
     Py_TYPE (self)->tp_free (self);
 
 done:
-    CPy_TRASHCAN_END (self)
+    CPy_TRASHCAN_END (self);
 }
 #endif
 
