diff -urN gtk-2.24.23.orig/gtk/gtkstatusicon.c gtk-2.24.23/gtk/gtkstatusicon.c
--- gtk-2.24.23.orig/gtk/gtkstatusicon.c	2014-03-05 20:56:42 -0800
+++ gtk-2.24.23/gtk/gtkstatusicon.c	2014-05-04 01:24:17 -0700
@@ -687,6 +687,25 @@
 
 static UINT taskbar_created_msg = 0;
 static GSList *status_icons = NULL;
+static UINT last_status_icon_id = -1;
+
+static GtkStatusIcon* find_status_icon (UINT id)
+{
+  GSList *rover;
+
+  for (rover = status_icons; rover != NULL; rover = rover->next)
+    {
+      GtkStatusIcon *status_icon = GTK_STATUS_ICON (rover->data);
+      GtkStatusIconPrivate *priv = status_icon->priv;
+      
+      if (priv->nid.uID == id)
+        {
+          return status_icon;
+        }
+    }
+
+  return NULL;
+}
 
 static LRESULT CALLBACK
 wndproc (HWND   hwnd,
@@ -704,7 +723,7 @@
 	  GtkStatusIconPrivate *priv = status_icon->priv;
 
 	  priv->nid.hWnd = hwnd;
-	  priv->nid.uID = GPOINTER_TO_UINT (status_icon);
+	  priv->nid.uID = ++last_status_icon_id;
 	  priv->nid.uCallbackMessage = WM_GTK_TRAY_NOTIFICATION;
 	  priv->nid.uFlags = NIF_MESSAGE;
 
@@ -748,7 +767,10 @@
 	buttondown0:
 	  bc = g_new (ButtonCallbackData, 1);
 	  bc->event = (GdkEventButton *) gdk_event_new (GDK_BUTTON_PRESS);
-	  bc->status_icon = GTK_STATUS_ICON (wparam);
+	  
+	  bc->status_icon = find_status_icon (wparam);
+	  g_assert (bc->status_icon != NULL);
+
 	  build_button_event (bc->status_icon->priv, bc->event, button);
 	  g_idle_add (button_callback, bc);
 	  break;
@@ -774,7 +796,10 @@
 	buttonup0:
 	  bc = g_new (ButtonCallbackData, 1);
 	  bc->event = (GdkEventButton *) gdk_event_new (GDK_BUTTON_RELEASE);
-	  bc->status_icon = GTK_STATUS_ICON (wparam);
+	  
+	  bc->status_icon = find_status_icon (wparam);
+	  g_assert (bc->status_icon != NULL);
+	  
 	  build_button_event (bc->status_icon->priv, bc->event, button);
 	  g_idle_add (button_callback, bc);
 	  break;
@@ -906,7 +931,7 @@
   memset (&priv->nid, 0, sizeof (priv->nid));
 
   priv->nid.hWnd = create_tray_observer ();
-  priv->nid.uID = GPOINTER_TO_UINT (status_icon);
+  priv->nid.uID = ++last_status_icon_id;
   priv->nid.uCallbackMessage = WM_GTK_TRAY_NOTIFICATION;
   priv->nid.uFlags = NIF_MESSAGE;
 
