diff -urN gtk-2.24.23.orig/gtk/gtkaccelgroup.c gtk-2.24.23/gtk/gtkaccelgroup.c
--- gtk-2.24.23.orig/gtk/gtkaccelgroup.c	2014-03-05 20:56:42 -0800
+++ gtk-2.24.23/gtk/gtkaccelgroup.c	2014-05-04 01:25:12 -0700
@@ -956,14 +956,13 @@
     GDK_ISO_First_Group, GDK_ISO_Last_Group,
     GDK_Mode_switch, GDK_Num_Lock, GDK_Multi_key,
     GDK_Scroll_Lock, GDK_Sys_Req, 
-    GDK_Tab, GDK_ISO_Left_Tab, GDK_KP_Tab,
+    GDK_ISO_Left_Tab, GDK_KP_Tab,
     GDK_First_Virtual_Screen, GDK_Prev_Virtual_Screen,
     GDK_Next_Virtual_Screen, GDK_Last_Virtual_Screen,
     GDK_Terminate_Server, GDK_AudibleBell_Enable,
     0
   };
   static const guint invalid_unmodified_vals[] = {
-    GDK_Up, GDK_Down, GDK_Left, GDK_Right,
     GDK_KP_Up, GDK_KP_Down, GDK_KP_Left, GDK_KP_Right,
     0
   };

