diff -ru librsvg-cur/build/win32/detectenv-msvc.mak librsvg-new/build/win32/detectenv-msvc.mak
--- librsvg-cur/build/win32/detectenv-msvc.mak	2017-11-28 17:06:10.000000000 +0100
+++ librsvg-new/build/win32/detectenv-msvc.mak	2019-11-30 17:24:17.301918300 +0100
@@ -42,8 +42,10 @@
 VSVER = 12
 !elseif $(VCVERSION) > 1899 && $(VCVERSION) < 1910
 VSVER = 14
-!elseif $(VCVERSION) > 1909 && $(VCVERSION) < 2000
+!elseif $(VCVERSION) > 1909 && $(VCVERSION) < 1920
 VSVER = 15
+!elseif $(VCVERSION) > 1919 && $(VCVERSION) < 2000
+VSVER = 16
 !else
 VSVER = 0
 !endif
@@ -51,7 +53,7 @@
 !if "$(VSVER)" == "0"
 MSG = ^
 This NMake Makefile set supports Visual Studio^
-9 (2008) through 14 (2015).  Your Visual Studio^
+9 (2008) through 16 (2019).  Your Visual Studio^
 version is not supported.
 !error $(MSG)
 !endif
Only in librsvg-new: config.h
