--- org/Makefile.vc        2017-01-23 19:27:19.473928700 +0100
+++ Makefile.vc      2017-01-23 19:25:57.354615200 +0100
@@ -8,7 +8,9 @@
 # in $(GLIB_PREFIX)\include\glib-2.0 and $(GLIB_PREFIX)\lib\glib-2.0\include
 # and its import library will be found in $(GLIB_PREFIX)\lib.

+!if "$(GLIB_PREFIX)" == ""
 GLIB_PREFIX = ..\vs$(VSVER)\$(PLAT)
+!endif

 # The items below this line should not be changed, unless one is maintaining
 # the NMake Makefiles.
