--- "pango-fixed\\pango-view\\meson.build"	2018-04-07 12:00:40.000000000 +0200
+++ "pango\\pango-view\\meson.build"	2018-08-23 23:14:03.143519000 +0200
@@ -39,6 +39,8 @@
                           '-DPACKAGE_VERSION="@0@"'.format(meson.project_version()),
                         ])
 
+
+if host_machine.system() != 'windows'
 help2man = find_program('help2man', required: false)
 if help2man.found()
   help2man_opts = [
@@ -56,3 +58,4 @@
                 install: true,
                 install_dir: join_paths(pango_datadir, 'man/man1'))
 endif
+endif
