Subject: [PATCH] Fix .pc location when CMAKE_INSTALL_LIBDIR not set
---
Index: CMakeLists.txt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	(revision 6effe2ec6bdc636573f9ea82bff3add632862333)
+++ b/CMakeLists.txt	(revision 4984c652db2a5c3c8d21c7338f8ab4874dd0096a)
@@ -3,6 +3,9 @@
 
 PROJECT(libarchive C)
 #
+# Include standard installation directories
+include(GNUInstallDirs)
+#
 SET(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/build/cmake")
 if(NOT CMAKE_RUNTIME_OUTPUT_DIRECTORY)
   set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${libarchive_BINARY_DIR}/bin)
Index: libarchive/CMakeLists.txt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/libarchive/CMakeLists.txt b/libarchive/CMakeLists.txt
--- a/libarchive/CMakeLists.txt	(revision 6effe2ec6bdc636573f9ea82bff3add632862333)
+++ b/libarchive/CMakeLists.txt	(revision 4984c652db2a5c3c8d21c7338f8ab4874dd0096a)
@@ -266,10 +266,6 @@
   SET_TARGET_PROPERTIES(archive_static PROPERTIES OUTPUT_NAME archive)
 ENDIF(NOT WIN32 OR CYGWIN OR NOT BUILD_SHARED_LIBS)
 
-if(NOT DEFINED CMAKE_INSTALL_LIBDIR)
-    set(CMAKE_INSTALL_LIBDIR "lib")
-endif()
-
 IF(ENABLE_INSTALL)
   # How to install the libraries
   IF(BUILD_SHARED_LIBS)
