--- origsrc/win-iconv-0.0.6/win_iconv.c	2012-11-21 21:01:04.557380400 -0600
+++ src/win-iconv-0.0.6/win_iconv.c	2012-11-21 20:53:28.648702600 -0600
@@ -21,6 +21,12 @@
 #define _strnicmp strncasecmp
 #endif
 
+#ifdef __GNUC__
+#define UNUSED __attribute__((unused))
+#else
+#define UNUSED
+#endif
+
 /* WORKAROUND: */
 #ifndef UNDER_CE
 #define GetProcAddressA GetProcAddress
@@ -115,7 +121,7 @@ static int win_iconv_open(rec_iconv_t *c
 static int win_iconv_close(iconv_t cd);
 static size_t win_iconv(iconv_t cd, char **inbuf, size_t *inbytesleft, char **outbuf, size_t *outbytesleft);
 
-static int load_mlang();
+static int load_mlang(void);
 static int make_csconv(const char *name, csconv_t *cv);
 static int name_to_codepage(const char *name);
 static uint utf16_to_ucs4(const ushort *wbuf);
@@ -685,7 +691,7 @@ static LCIDTORFC1766A LcidToRfc1766A;
 static RFC1766TOLCIDA Rfc1766ToLcidA;
 
 static int
-load_mlang()
+load_mlang(void)
 {
     HMODULE h;
     if (ConvertINetString != NULL)
@@ -765,7 +771,7 @@ win_iconv_open(rec_iconv_t *cd, const ch
 }
 
 static int
-win_iconv_close(iconv_t cd)
+win_iconv_close(iconv_t cd UNUSED)
 {
     return 0;
 }
@@ -1259,7 +1265,7 @@ find_imported_module_by_funcname(HMODULE
 #endif
 
 static int
-sbcs_mblen(csconv_t *cv, const uchar *buf, int bufsize)
+sbcs_mblen(csconv_t *cv UNUSED, const uchar *buf UNUSED, int bufsize UNUSED)
 {
     return 1;
 }
@@ -1296,7 +1302,7 @@ mbcs_mblen(csconv_t *cv, const uchar *bu
 }
 
 static int
-utf8_mblen(csconv_t *cv, const uchar *buf, int bufsize)
+utf8_mblen(csconv_t *cv UNUSED, const uchar *buf, int bufsize)
 {
     int len = 0;
 
@@ -1315,7 +1321,7 @@ utf8_mblen(csconv_t *cv, const uchar *bu
 }
 
 static int
-eucjp_mblen(csconv_t *cv, const uchar *buf, int bufsize)
+eucjp_mblen(csconv_t *cv UNUSED, const uchar *buf, int bufsize)
 {
     if (buf[0] < 0x80) /* ASCII */
         return 1;
