diff -urpN win-iconv-0.0.6.orig/iconv.h win-iconv-0.0.6/iconv.h
--- win-iconv-0.0.6.orig/iconv.h	2012-09-19 06:00:04.000000000 -0700
+++ win-iconv-0.0.6/iconv.h	2015-03-15 21:33:11.119696700 -0700
@@ -7,7 +7,7 @@ extern "C" {
 typedef void* iconv_t;
 iconv_t iconv_open(const char *tocode, const char *fromcode);
 int iconv_close(iconv_t cd);
-size_t iconv(iconv_t cd, char **inbuf, size_t *inbytesleft, char **outbuf, size_t *outbytesleft);
+size_t iconv(iconv_t cd, const char **inbuf, size_t *inbytesleft, char **outbuf, size_t *outbytesleft);
 #ifdef __cplusplus
 }
 #endif
diff -urpN win-iconv-0.0.6.orig/win_iconv.c win-iconv-0.0.6/win_iconv.c
--- win-iconv-0.0.6.orig/win_iconv.c	2015-03-15 21:32:01.447795100 -0700
+++ win-iconv-0.0.6/win_iconv.c	2015-03-15 21:33:42.179820000 -0700
@@ -53,7 +53,7 @@ typedef void* iconv_t;
 
 iconv_t iconv_open(const char *tocode, const char *fromcode);
 int iconv_close(iconv_t cd);
-size_t iconv(iconv_t cd, char **inbuf, size_t *inbytesleft, char **outbuf, size_t *outbytesleft);
+size_t iconv(iconv_t cd, const char **inbuf, size_t *inbytesleft, char **outbuf, size_t *outbytesleft);
 
 /* libiconv interface for vim */
 #if defined(MAKE_DLL)
@@ -71,7 +71,7 @@ typedef struct rec_iconv_t rec_iconv_t;
 
 typedef iconv_t (*f_iconv_open)(const char *tocode, const char *fromcode);
 typedef int (*f_iconv_close)(iconv_t cd);
-typedef size_t (*f_iconv)(iconv_t cd, char **inbuf, size_t *inbytesleft, char **outbuf, size_t *outbytesleft);
+typedef size_t (*f_iconv)(iconv_t cd, const char **inbuf, size_t *inbytesleft, char **outbuf, size_t *outbytesleft);
 typedef int* (*f_errno)(void);
 typedef int (*f_mbtowc)(csconv_t *cv, const uchar *buf, int bufsize, ushort *wbuf, int *wbufsize);
 typedef int (*f_wctomb)(csconv_t *cv, ushort *wbuf, int wbufsize, uchar *buf, int bufsize);
@@ -113,7 +113,7 @@ struct rec_iconv_t {
 
 static int win_iconv_open(rec_iconv_t *cd, const char *tocode, const char *fromcode);
 static int win_iconv_close(iconv_t cd);
-static size_t win_iconv(iconv_t cd, char **inbuf, size_t *inbytesleft, char **outbuf, size_t *outbytesleft);
+static size_t win_iconv(iconv_t cd, const char **inbuf, size_t *inbytesleft, char **outbuf, size_t *outbytesleft);
 
 static int load_mlang(void);
 static int make_csconv(const char *name, csconv_t *cv);
@@ -744,7 +744,7 @@ iconv_close(iconv_t _cd)
 }
 
 size_t
-iconv(iconv_t _cd, char **inbuf, size_t *inbytesleft, char **outbuf, size_t *outbytesleft)
+iconv(iconv_t _cd, const char **inbuf, size_t *inbytesleft, char **outbuf, size_t *outbytesleft)
 {
     rec_iconv_t *cd = (rec_iconv_t *)_cd;
     size_t r = cd->iconv(cd->cd, inbuf, inbytesleft, outbuf, outbytesleft);
@@ -771,7 +771,7 @@ win_iconv_close(iconv_t cd UNUSED)
 }
 
 static size_t
-win_iconv(iconv_t _cd, char **inbuf, size_t *inbytesleft, char **outbuf, size_t *outbytesleft)
+win_iconv(iconv_t _cd, const char **inbuf, size_t *inbytesleft, char **outbuf, size_t *outbytesleft)
 {
     rec_iconv_t *cd = (rec_iconv_t *)_cd;
     ushort wbuf[MB_CHAR_MAX]; /* enough room for one character */
@@ -1939,7 +1939,7 @@ main(int argc, char **argv)
     int i;
     char inbuf[BUFSIZ];
     char outbuf[BUFSIZ];
-    char *pin;
+    const char *pin;
     char *pout;
     size_t inbytesleft;
     size_t outbytesleft;
diff -urpN win-iconv-0.0.6.orig/win_iconv_test.c win-iconv-0.0.6/win_iconv_test.c
--- win-iconv-0.0.6.orig/win_iconv_test.c	2012-10-21 05:53:04.000000000 -0700
+++ win-iconv-0.0.6/win_iconv_test.c	2015-03-15 21:32:34.928848700 -0700
@@ -90,7 +90,7 @@ void
 test(const char *from, const char *fromstr, int fromsize, const char *to, const char *tostr, int tosize, int errcode, int bufsize, int line)
 {
     char outbuf[BUFSIZ];
-    char *pin;
+    const char *pin;
     char *pout;
     size_t inbytesleft;
     size_t outbytesleft;
@@ -125,7 +125,7 @@ test(const char *from, const char *froms
 
     errno = 0;
 
-    pin = (char *)fromstr;
+    pin = fromstr;
     pout = outbuf;
     inbytesleft = fromsize;
     outbytesleft = bufsize;
