Subject: [PATCH] Remove appstream
---
Index: src/adw-about-window.c
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/adw-about-window.c b/src/adw-about-window.c
--- a/src/adw-about-window.c	(revision 5808d27bd19fd16af6bafdb35c64c4a492daeaf5)
+++ b/src/adw-about-window.c	(date 1773605476407)
@@ -6,7 +6,9 @@
 
 #include "config.h"
 #include <glib/gi18n-lib.h>
+#ifndef G_OS_WIN32
 #include <appstream.h>
+#endif
 
 #include "adw-about-window.h"
 
@@ -427,12 +429,14 @@
     g_idle_add_once ((GSourceOnceFunc) legal_showing_idle_cb, self);
 }
 
+#ifndef G_OS_WIN32
 static gboolean
 get_release_for_version (AsRelease  *rel,
                          const char *version)
 {
   return !g_strcmp0 (as_release_get_version (rel), version);
 }
+#endif
 
 static void
 update_credits_legal_group (AdwAboutWindow *self)
@@ -1961,6 +1965,7 @@
   return g_object_new (ADW_TYPE_ABOUT_WINDOW, NULL);
 }
 
+#ifndef G_OS_WIN32
 /**
  * adw_about_window_new_from_appdata:
  * @resource_path: The resource to use
@@ -2143,6 +2148,7 @@
 
   return GTK_WIDGET (self);
 }
+#endif
 
 /**
  * adw_about_window_get_application_icon:
@@ -3479,6 +3485,7 @@
   gtk_window_present (GTK_WINDOW (window));
 }
 
+#ifndef G_OS_WIN32
 /**
  * adw_show_about_window_from_appdata: (skip)
  * @parent: (nullable): the parent top-level window
@@ -3518,3 +3525,4 @@
 
   gtk_window_present (GTK_WINDOW (window));
 }
+#endif
Index: src/adw-about-window.h
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/adw-about-window.h b/src/adw-about-window.h
--- a/src/adw-about-window.h	(revision 5808d27bd19fd16af6bafdb35c64c4a492daeaf5)
+++ b/src/adw-about-window.h	(date 1773605476413)
@@ -25,9 +25,11 @@
 ADW_DEPRECATED_IN_1_6_FOR(adw_about_dialog_new)
 GtkWidget *adw_about_window_new (void) G_GNUC_WARN_UNUSED_RESULT;
 
+#ifndef G_OS_WIN32
 ADW_DEPRECATED_IN_1_6_FOR(adw_about_dialog_new_from_appdata)
 GtkWidget *adw_about_window_new_from_appdata (const char *resource_path,
                                               const char *release_notes_version) G_GNUC_WARN_UNUSED_RESULT;
+#endif
 
 ADW_DEPRECATED_IN_1_6_FOR(adw_about_dialog_get_application_name)
 const char *adw_about_window_get_application_name (AdwAboutWindow *self);
@@ -176,11 +178,13 @@
                             const char *first_property_name,
                             ...) G_GNUC_NULL_TERMINATED;
 
+#ifndef G_OS_WIN32
 ADW_DEPRECATED_IN_1_6_FOR(adw_show_about_dialog_from_appdata)
 void adw_show_about_window_from_appdata (GtkWindow  *parent,
                                          const char *resource_path,
                                          const char *release_notes_version,
                                          const char *first_property_name,
                                          ...) G_GNUC_NULL_TERMINATED;
+#endif
 
 G_END_DECLS
Index: src/meson.build
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/meson.build b/src/meson.build
--- a/src/meson.build	(revision 5808d27bd19fd16af6bafdb35c64c4a492daeaf5)
+++ b/src/meson.build	(date 1773605476399)
@@ -327,14 +327,24 @@
 
 install_headers(src_headers, subdir: libadwaita_header_subdir)
 
-libadwaita_deps = [
-  glib_dep,
-  fribidi_dep,
-  gio_dep,
-  gtk_dep,
-  appstream_dep,
-  cc.find_library('m', required: false),
-]
+if target_system != 'windows'
+  libadwaita_deps = [
+    glib_dep,
+    fribidi_dep,
+    gio_dep,
+    gtk_dep,
+    appstream_dep,
+    cc.find_library('m', required: false),
+  ]
+else
+  libadwaita_deps = [
+    glib_dep,
+    fribidi_dep,
+    gio_dep,
+    gtk_dep,
+    cc.find_library('m', required: false),
+  ]
+endif
 
 libadwaita_public_deps = [
   gio_dep,
Index: src/adw-about-dialog.c
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/adw-about-dialog.c b/src/adw-about-dialog.c
--- a/src/adw-about-dialog.c	(revision 5808d27bd19fd16af6bafdb35c64c4a492daeaf5)
+++ b/src/adw-about-dialog.c	(date 1773606869511)
@@ -7,7 +7,9 @@
 
 #include "config.h"
 #include <glib/gi18n-lib.h>
+#ifndef G_OS_WIN32
 #include <appstream.h>
+#endif
 
 #include "adw-about-dialog.h"
 
@@ -318,14 +320,18 @@
 
   guint legal_showing_idle_id;
 
+  #ifndef G_OS_WIN32
   char *appdata_resource_path;
+  #endif
 };
 
 G_DEFINE_FINAL_TYPE (AdwAboutDialog, adw_about_dialog, ADW_TYPE_DIALOG)
 
 enum {
   PROP_0,
+#ifndef G_OS_WIN32
   PROP_APPDATA_RESOURCE_PATH,
+#endif
   PROP_APPLICATION_ICON,
   PROP_APPLICATION_NAME,
   PROP_DEVELOPER_NAME,
@@ -437,12 +443,14 @@
   self->legal_showing_idle_id = 0;
 }
 
+#ifndef G_OS_WIN32
 static gboolean
 get_release_for_version (AsRelease  *rel,
                          const char *version)
 {
   return !g_strcmp0 (as_release_get_version (rel), version);
 }
+#endif
 
 static void
 update_credits_legal_group (AdwAboutDialog *self)
@@ -1074,6 +1082,7 @@
                           has_support_url || has_issue_url || has_debug_info);
 }
 
+#ifndef G_OS_WIN32
 static void
 populate_from_appdata (AdwAboutDialog *self)
 {
@@ -1212,6 +1221,7 @@
   g_free (application_id);
   g_free (appdata_uri);
 }
+#endif
 
 static void
 adw_about_dialog_get_property (GObject    *object,
@@ -1222,9 +1232,11 @@
   AdwAboutDialog *self = ADW_ABOUT_DIALOG (object);
 
   switch (prop_id) {
+  #ifndef G_OS_WIN32
   case PROP_APPDATA_RESOURCE_PATH:
     g_value_set_string (value, adw_about_dialog_get_appdata_resource_path (self));
     break;
+  #endif
   case PROP_APPLICATION_ICON:
     g_value_set_string (value, adw_about_dialog_get_application_icon (self));
     break;
@@ -1299,9 +1311,11 @@
   AdwAboutDialog *self = ADW_ABOUT_DIALOG (object);
 
   switch (prop_id) {
+  #ifndef G_OS_WIN32
   case PROP_APPDATA_RESOURCE_PATH:
     g_set_str (&self->appdata_resource_path, g_value_get_string (value));
     break;
+  #endif
   case PROP_APPLICATION_ICON:
     adw_about_dialog_set_application_icon (self, g_value_get_string (value));
     break;
@@ -1374,8 +1388,11 @@
 
   G_OBJECT_CLASS (adw_about_dialog_parent_class)->constructed (object);
 
+
+  #ifndef G_OS_WIN32
   if (self->appdata_resource_path)
     populate_from_appdata (self);
+  #endif
 }
 
 static void
@@ -1417,7 +1434,9 @@
   g_free (self->license);
   g_slist_free_full (self->legal_sections, (GDestroyNotify) free_legal_section);
 
+  #ifndef G_OS_WIN32
   g_free (self->appdata_resource_path);
+  #endif
 
   G_OBJECT_CLASS (adw_about_dialog_parent_class)->finalize (object);
 }
@@ -1524,6 +1543,7 @@
   object_class->get_property = adw_about_dialog_get_property;
   object_class->set_property = adw_about_dialog_set_property;
 
+  #ifndef G_OS_WIN32
   /**
    * AdwAboutDialog:appdata-resource-path:
    *
@@ -1542,6 +1562,7 @@
     g_param_spec_string ("appdata-resource-path", NULL, NULL,
                          NULL,
                          G_PARAM_READWRITE | G_PARAM_CONSTRUCT_ONLY | G_PARAM_STATIC_STRINGS);
+  #endif
 
   /**
    * AdwAboutDialog:application-icon:
@@ -2128,6 +2149,7 @@
   return g_object_new (ADW_TYPE_ABOUT_DIALOG, NULL);
 }
 
+#ifndef G_OS_WIN32
 /**
  * adw_about_dialog_new_from_appdata:
  * @resource_path: The resource to use
@@ -2173,6 +2195,7 @@
                        "release-notes-version", release_notes_version,
                        NULL);
 }
+#endif
 
 /**
  * adw_about_dialog_get_application_icon:
@@ -2219,6 +2242,7 @@
   g_object_notify_by_pspec (G_OBJECT (self), props[PROP_APPLICATION_ICON]);
 }
 
+#ifndef G_OS_WIN32
 /**
  * adw_about_dialog_get_appdata_resource_path:
  * @self: an about dialog
@@ -2236,6 +2260,7 @@
 
   return self->appdata_resource_path;
 }
+#endif
 
 /**
  * adw_about_dialog_get_application_name:
@@ -3558,6 +3583,7 @@
   adw_dialog_present (dialog, parent);
 }
 
+#ifndef G_OS_WIN32
 /**
  * adw_show_about_dialog_from_appdata: (skip)
  * @parent: the parent widget
@@ -3595,4 +3621,5 @@
 
   adw_dialog_present (dialog, parent);
 }
+#endif
 
Index: meson.build
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/meson.build b/meson.build
--- a/meson.build	(revision 5808d27bd19fd16af6bafdb35c64c4a492daeaf5)
+++ b/meson.build	(date 1773605476399)
@@ -40,15 +40,18 @@
 gio_dep = dependency('gio-2.0', version: glib_min_version)
 gtk_dep = dependency('gtk4', version: gtk_min_version)
 fribidi_dep = dependency('fribidi')
-appstream_dep = dependency('appstream',
-  fallback : ['appstream', 'appstream_dep'],
-  default_options : [
-    'systemd=false', 'apidocs=false', 'install-docs=false',
-    'stemming=false', 'svg-support=false', 'gir=false',
-  ],
-)
+
+target_system = host_machine.system()
+if target_system != 'windows'
+  appstream_dep = dependency('appstream',
+    fallback : ['appstream', 'appstream_dep'],
+    default_options : [
+      'systemd=false', 'apidocs=false', 'install-docs=false',
+      'stemming=false', 'svg-support=false', 'gir=false',
+    ],
+  )
+endif
 
-target_system = host_machine.system()
 if target_system == 'darwin'
   appleframework_modules = [
     'AppKit',
Index: src/adw-about-dialog.h
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/adw-about-dialog.h b/src/adw-about-dialog.h
--- a/src/adw-about-dialog.h	(revision 5808d27bd19fd16af6bafdb35c64c4a492daeaf5)
+++ b/src/adw-about-dialog.h	(date 1773605476410)
@@ -27,12 +27,13 @@
 ADW_AVAILABLE_IN_1_5
 AdwDialog *adw_about_dialog_new (void) G_GNUC_WARN_UNUSED_RESULT;
 
+#ifndef G_OS_WIN32
 ADW_AVAILABLE_IN_1_5
 AdwDialog *adw_about_dialog_new_from_appdata (const char *resource_path,
                                               const char *release_notes_version) G_GNUC_WARN_UNUSED_RESULT;
-
 ADW_AVAILABLE_IN_1_9
 const char *adw_about_dialog_get_appdata_resource_path (AdwAboutDialog *self);
+#endif
 
 ADW_AVAILABLE_IN_1_5
 const char *adw_about_dialog_get_application_name (AdwAboutDialog *self);
@@ -187,11 +188,13 @@
                             const char *first_property_name,
                             ...) G_GNUC_NULL_TERMINATED;
 
+#ifndef G_OS_WIN32
 ADW_AVAILABLE_IN_1_5
 void adw_show_about_dialog_from_appdata (GtkWidget  *parent,
                                          const char *resource_path,
                                          const char *release_notes_version,
                                          const char *first_property_name,
                                          ...) G_GNUC_NULL_TERMINATED;
+#endif
 
 G_END_DECLS
