diff --git a/libgxps/gxps-page.c b/libgxps/gxps-page.c
index 3333505..ed751bf 100644
--- a/libgxps/gxps-page.c
+++ b/libgxps/gxps-page.c
@@ -993,6 +993,7 @@ render_end_element (GMarkupParseContext  *context,
 		cairo_matrix_init_identity (&font_matrix);
 		cairo_matrix_scale (&font_matrix, glyphs->em_size, glyphs->em_size);
 		cairo_get_matrix (ctx->cr, &ctm);
+		cairo_matrix_scale (&ctm, glyphs->em_size, glyphs->em_size);
 
                 /* italics is 20 degrees slant.  0.342 = sin(20 deg) */
                 if (glyphs->italic)
