From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Mark Jansen <mark.jansen@reactos.org>
Date: Thu, 16 Nov 2017 20:51:50 +0100
Subject: [PATCH] GDK W32: Remove WS_EX_LAYERED from an opaque window

---
 gdk/win32/gdkwindow-win32.c | 23 ++++++++++++++++++-----
 1 file changed, 18 insertions(+), 5 deletions(-)

diff --git a/gdk/win32/gdkwindow-win32.c b/gdk/win32/gdkwindow-win32.c
index a82dc0f..da18aa7 100644
--- a/gdk/win32/gdkwindow-win32.c
+++ b/gdk/win32/gdkwindow-win32.c
@@ -3939,6 +3939,7 @@ gdk_window_set_opacity (GdkWindow *window,
 			gdouble    opacity)
 {
   LONG exstyle;
+  BYTE newOpacity;
   typedef BOOL (WINAPI *PFN_SetLayeredWindowAttributes) (HWND, COLORREF, BYTE, DWORD);
   PFN_SetLayeredWindowAttributes setLayeredWindowAttributes = NULL;
 
@@ -3953,12 +3954,24 @@ gdk_window_set_opacity (GdkWindow *window,
   else if (opacity > 1)
     opacity = 1;
 
+  newOpacity = opacity * 0xff;
+
   exstyle = GetWindowLong (GDK_WINDOW_HWND (window), GWL_EXSTYLE);
 
-  if (!(exstyle & WS_EX_LAYERED))
-    SetWindowLong (GDK_WINDOW_HWND (window),
-		    GWL_EXSTYLE,
-		    exstyle | WS_EX_LAYERED);
+  if (newOpacity != 255)
+  {
+    if (!(exstyle & WS_EX_LAYERED))
+      SetWindowLong (GDK_WINDOW_HWND (window),
+		      GWL_EXSTYLE,
+		      exstyle | WS_EX_LAYERED);
+  }
+  else
+  {
+    if ((exstyle & WS_EX_LAYERED))
+      SetWindowLong (GDK_WINDOW_HWND (window),
+		      GWL_EXSTYLE,
+		      exstyle & ~WS_EX_LAYERED);
+  }
 
   setLayeredWindowAttributes = 
     (PFN_SetLayeredWindowAttributes)GetProcAddress (GetModuleHandle ("user32.dll"), "SetLayeredWindowAttributes");
@@ -3967,7 +3980,7 @@ gdk_window_set_opacity (GdkWindow *window,
     {
       API_CALL (setLayeredWindowAttributes, (GDK_WINDOW_HWND (window),
 					     0,
-					     opacity * 0xff,
+					     newOpacity,
 					     LWA_ALPHA));
     }
 }
-- 
2.12.2.windows.1

