diff -urpN gtk-2.24.24.orig/gdk/win32/gdkwindow-win32.c gtk-2.24.24/gdk/win32/gdkwindow-win32.c
--- gtk-2.24.24.orig/gdk/win32/gdkwindow-win32.c	Mon Mar 17 18:19:03 2014
+++ gtk-2.24.24/gdk/win32/gdkwindow-win32.c	Sun Sep  7 16:51:16 2014
@@ -1182,6 +1182,9 @@ show_window_internal (GdkWindow *window,
 
 	  x = center_on_rect.left + ((center_on_rect.right - center_on_rect.left) - (window_rect.right - window_rect.left)) / 2;
 	  y = center_on_rect.top + ((center_on_rect.bottom - center_on_rect.top) - (window_rect.bottom - window_rect.top)) / 2;
+
+	  x -= _gdk_offset_x;
+	  y -= _gdk_offset_y;
 	}
 
       API_CALL (SetWindowPos, (GDK_WINDOW_HWND (window), NULL,
