Subject: [PATCH] fix-python-not-found
---
Index: meson.build
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/meson.build b/meson.build
--- a/meson.build	(revision 2b3b3856564ea702cf244ed85fb06a654a95259c)
+++ b/meson.build	(date 1743462947854)
@@ -49,7 +49,7 @@
 
 python_executable = is_msvc ? 'python' : 'python3'
 
-python3 = find_program(python_executable, version: '>=3.7')
+python3 = import('python').find_installation()
 pkgconfig = import('pkgconfig')
 common_pkgconfig_vars = [
   'doxytagfile=${docdir}/reference/@0@.tag'.format(glibmm_pcname),
