diff -ru pkg-config/libpkgconf/bsdstubs.c pkg-config-vs2013/libpkgconf/bsdstubs.c
--- pkg-config/libpkgconf/bsdstubs.c	2018-06-14 16:37:45.356817100 +0200
+++ pkg-config-vs2013/libpkgconf/bsdstubs.c	2018-06-14 15:59:29.448953600 +0200
@@ -22,6 +22,7 @@
 #include <string.h>

 #include <libpkgconf/bsdstubs.h>
+#include <libpkgconf/libpkgconf.h>
 #include <libpkgconf/config.h>

 #ifndef HAVE_STRLCPY
diff -ru pkg-config/libpkgconf/libpkgconf.h pkg-config-vs2013/libpkgconf/libpkgconf.h
--- pkg-config/libpkgconf/libpkgconf.h	2018-06-14 16:37:45.357815200 +0200
+++ pkg-config-vs2013/libpkgconf/libpkgconf.h	2018-06-14 16:36:56.034105600 +0200
@@ -20,6 +20,14 @@
 #include <stdarg.h>
 #include <stddef.h>
 #include <stdbool.h>
+
+/* Macro for building with vs2013 */
+#if defined _MSC_VER && _MSC_VER <= 1800
+#define inline		__inline
+#define snprintf	_snprintf
+#define __func__	__FUNCTION__
+#endif
+
 #include <libpkgconf/libpkgconf-api.h>
 #include <libpkgconf/iter.h>
 #include <libpkgconf/bsdstubs.h>
